Fix detection of mobile browsers

c727 opened the pull request:
## Description:

Mobile browsers have "Mobile" in their family name like "Chrome Mobile".
fix by @mhofman

**Related issue (if applicable):** fixes #12068

## Checklist:
  - [x] The code change is tested and works locally.

If the code does not interact with devices:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**

Commit:
Fix detection of mobile browsers

fabaff committed:
Move break

fabaff reviewed (approved):
Thanks :bird:

balloob reviewed (changes requested):
I think that it's better to just make new entries for the mobile browsers instead of trying to "hack" their name to match their desktop variants.

Commit:
Add Mobile browsers to list

`Mobile Safari` is covered by iOS and `Opera Mini` does not support es6
... And Edge doesn't even work on desktop for me...

balloob commented:
What about Safari?

c727 commented:
the comment was not copied here:
`Mobile Safari` is covered by iOS and `Opera Mini` does not support es6
... And Edge doesn't even work on desktop for me...

let me charge my tablet and I will also test Edge (Desktoo) again

c727 commented:
...I'm running Edge 38 Desktop and have the same issues like here (es5 and latest) https://github.com/home-assistant/home-assistant-polymer/issues/851

mhofman commented:
@balloob, why do you think it's a hack to consider mobile browser the equivalent of their desktop counterpart.
I do not know of any browser that runs a different JS engine in their desktop and mobile versions.

Further, none have different JS syntax support, which is what should be important here. Once the JS is parsed, API differences (including core JS functions) can be polyfilled dynamically.

balloob commented:
It's hacky because it would also capture Mobile Edge etc, which we were not targeting here.
